src/serialized-analysis.ts

```diff
@@ -0,0 +1,425 @@
+export interface SerializedAnalysis { packages: Package[]; }
+
+export interface Package {
+  /** The name of the package, like `paper-button` */
+  name: string;
+  /** The version, extracted from package.json, bower.json, etc as available. */
+  version: string;
+
+  /** The npm metadata of the package, if any. */
+  npmPackage?: NpmPackage;
+
+  /** The bower metadata of the package, if any. */
+  bowerMetadata?: BowerMetadata;
+
+  /** Elements found inside the package. */
+  elements: Element[];
+}
+
+export interface Element {
+  /**
+   * The path, relative to the base directory of the package.
+   *
+   * e.g. `paper-input.html` or `app-toolbar/app-toolbar.html` (given that
+   * app-toolbar lives in the app-layout package).
+   */
+  path: string;
+
+  /** The tagname that the element registers itself as. e.g. `paper-input` */
+  tagname: string;
+
+  /** A markdown description for the element. */
+  description: string;
+
+  /**
+   * Paths, relative to `this.path` to demo pages for the element.
+   *
+   * e.g. `['demos/index.html', 'demos/extended.html']`
+   */
+  demos: string[];
+
+  /**
+   * The tagname that the element extends, if any. The value of the `extends`
+   * option that's passed into `customElements.define`.
+   *
+   * e.g. `input`, `paper-button`, `my-super-element`
+   */
+  extends?: string;
+
+  /**
+   * The class name for this element.
+   *
+   * e.g. `MyElement`, `Polymer.PaperInput`
+   */
+  classname: string;
+
+  /**
+   * The class that this element extends.
+   *
+   * This is non-optional, as every custom element must have HTMLElement in
+   * its prototype change.
+   *
+   * e.g. `HTMLElement`, `HTMLInputElement`, `MyNamespace.MyBaseElement`
+   */
+  superclass: string;
+
+  /** The attributes that this element is known to understand. */
+  attributes?: Attribute[];
+
+  /** The properties that this element has. */
+  properties?: Property[];
+
+  /** The events that this element fires. */
+  events?: Event[];
+
+  /** The shadow dom content slots that this element accepts. */
+  'slots': Slot[];
+
+  /** Information useful for styling the element and its children. */
+  styling: {
+
+    /** CSS Classes that produce built-in custom styling for the element. */
+    classes: {
```

Comment:
I'm not sure this should just be classes. An element could have a `:host([height=5])` rule. Maybe just selectors?